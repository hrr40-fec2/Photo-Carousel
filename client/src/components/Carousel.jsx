import React from 'react';
import Slider from 'react-slick';
import ImageList from './ImageList.jsx';
import styled from 'styled-components';


function NextArrow(props) {
  const { className, style, onClick } = props;
  return (
    <div
      className={className}
      style={{ ...style, display: "block", background: "grey" }}
      onClick={onClick}
    />
  );
}

function PrevArrow(props) {
  const { className, style, onClick } = props;
  return (
    <div
      className={className}
      style={{ ...style, display: "block", background: "grey" }}
      onClick={onClick}
    />
  );
}

var Wrapper = styled.div`
  width: 50%;
  height 50%;
`;

var Slides = styled.div`
  display: flex;
  justify-content: center;
`;



export default class Carousel extends React.Component {

  render() {
    var settings = {
      dots: true,
      infinite: true,
      speed: 500,
      slidesToShow: 6,
      slidesToScroll: 2,
      centerPadding: 30,
      className: 'slides',
      nextArrow: <NextArrow />,
      prevArrow: <PrevArrow />
    };

    var imageStyle = {
      width: '100px',
      height: '100px'
    };

    return (
      <Wrapper>
        <h1>Photos</h1>
        <Slider {...settings}>
          {this.props.photos.map((photo, index) => {
            return (
              <Slides key={index}>
                <img src={photo} style={imageStyle} ></img>
              </Slides>
            )
          })}
        </Slider>
      </Wrapper>
    );
  }
}