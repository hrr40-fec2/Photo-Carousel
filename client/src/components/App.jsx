import React from 'react';
import Carousel from './Carousel.jsx';
import $ from 'jquery';


export default class App extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      images: []
    }
  }


  render() {
    return (
      <Carousel />
    )
  }
};