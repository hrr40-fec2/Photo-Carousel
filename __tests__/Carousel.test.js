import { shallow } from 'enzyme';
import React from 'react';
import Tests from '../client/src/setupTests.js';
import Carousel from '../client/src/components/Carousel.jsx';
import App from '../client/src/components/App.jsx';

describe('Carousel component', () => {
  //it should hold 25 photos
  it('should hold 25 photos', () => {

  });

  //renders a Carousel component
  it('renders correctly', () => {
    const images = ['randomImage1', 'randomImage2'];
    const carousel = shallow(<Carousel images={images} />);
    expect(carousel).toMatchSnapshot();
  });

  it('should hold images', () => {
    const images = ['randomImage1', 'randomImage2'];
    const carousel = shallow(<Carousel photos={images} />)
    expect(carousel).toMatchSnapshot();
  });


});