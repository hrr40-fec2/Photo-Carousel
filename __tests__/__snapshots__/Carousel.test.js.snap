// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel component should hold images 1`] = `ShallowWrapper {}`;
